pkg/aws/eni/instances.go

```diff
@@ -278,6 +279,51 @@ func (m *InstancesManager) UpdateENI(instanceID string, eni *eniTypes.ENI) {
 	m.mutex.Unlock()
 }
 
+func (m *InstancesManager) AddIPsToENI(instanceID string, eniID string, ips []string) {
+	m.modifyIPsToENI(instanceID, eniID, ips, true)
+}
+
+func (m *InstancesManager) RemoveIPsFromENI(instanceID string, eniID string, ips []string) {
+	m.modifyIPsToENI(instanceID, eniID, ips, false)
+}
+
+func (m *InstancesManager) modifyIPsToENI(instanceID string, eniID string, ips []string, isAdd bool) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	ifaces, ok := m.instances.GetInterface(instanceID, eniID)
+	if !ok {
+		log.WithFields(logrus.Fields{
+			"instance": instanceID,
+			"eni":      eniID,
+		}).Warning("ENI not found")
+		return
+	}
+
+	eniIntf := ifaces.Resource.DeepCopyInterface()
+	eni, ok := eniIntf.(*eniTypes.ENI)
+	if !ok {
+		log.WithFields(logrus.Fields{
+			"instance": instanceID,
+			"eni":      eniID,
+		}).Warning("Unexpected resource type, expected *eniTypes.ENI")
+		return
+	}
+	if isAdd {
+		for _, ip := range ips {
+			if !slices.Contains(eni.Addresses, ip) {
+				eni.Addresses = append(eni.Addresses, ip)
+			}
+		}
+	} else {
```

Comment:
@gandro I looked into it today trying to implement the remove IP and update PD.

I actually found modifying the exisitng PR is easier than the new PR. Updating PD is not really necessary since that will not affect the ip address calculation. I think we are ok for the stale PD info and wait for the async update.

I just refactor the function here for both add and remove, and remove the ips in the instance map(I dont need to modify our ec2 api to for unassign since aws doesn't return those values back to us).

Reply:
Sure, sounds good to me!